sources/packages/backend/apps/api/src/route-controllers/config/_tests_/e2e/config.controller.getConfig.e2e-spec.ts

```diff
@@ -0,0 +1,61 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+
+import { createTestingConfigModule } from "../../../../testHelpers";
+import * as request from "supertest";
+import { ConfigAPIOutDTO } from "../../models/config.dto";
+
+describe("ConfigController(e2e)-getConfig", () => {
+  let app: INestApplication;
+  const originalEnv = process.env;
+  const fakeEnvVariables = {
+    KEYCLOAK_AUTH_URL: "https://keycloak-fake-url",
+    KEYCLOAK_REALM: "keycloak_fake_realm",
+    KEYCLOAK_CLIENT_STUDENT: "student",
+    KEYCLOAK_CLIENT_INSTITUTION: "institution",
+    KEYCLOAK_CLIENT_AEST: "aest",
+    KEYCLOAK_CLIENT_SUPPORTING_USERS: "supporting-users",
+    SITE_MINDER_LOGOUT_URL: "https://fake-siteminder-logout-url",
+  };
+
+  beforeAll(async () => {
+    jest.resetModules();
+    process.env = {
+      ...originalEnv,
+      ...fakeEnvVariables,
+    };
+    const { nestApplication } = await createTestingConfigModule();
+    app = nestApplication;
+  });
+
+  it("Should return config values", async () => {
+    // Arrange Act/Assert
+    await request(app.getHttpServer())
+      .get("/config")
+      .expect(HttpStatus.OK)
+      .then((response) => {
```

Comment:
The expected can be simplified as shown below (please change the hard-code variables to the declared ones if you wish). This also will ensure that the right amount of info is returned and that there is nothing unexpected leaking from the API response.
```ts
it("Should return config values", async () => {
    // Arrange Act/Assert
    await request(app.getHttpServer())
      .get("/config")
      .expect(HttpStatus.OK)
      .expect({
        auth: {
          url: "https://keycloak-fake-url",
          realm: "keycloak_fake_realm",
          clientIds: {
            student: "student",
            institution: "institution",
            aest: "aest",
            supportingUsers: "supporting-users",
          },
          externalSiteMinderLogoutUrl: "https://fake-siteminder-logout-url",
        },
      });
  });
```